pkg/upgrade/control_plane.go

```diff
@@ -974,7 +989,7 @@ func (u *ControlPlaneUpgrader) updateBootstrapConfig(replacementKey ctrlclient.O
 	return nil
 }
 
-func (u *ControlPlaneUpgrader) resourceExists(ref v1.ObjectReference) (bool, error) {
+func (u *ControlPlaneUpgrader) resourceExists(ref v1.ObjectReference) (bool, *unstructured.Unstructured, error) {
```

Comment:
```suggestion
func (u *ControlPlaneUpgrader) getFromReference(ref v1.ObjectReference) (*unstructured.Unstructured, error) {
```

Do we need the boolean field here anymore? It seems we can just rely on the field being nil and do the check for `apierrors.IsNotFound(err)` outside?

Reply:
It isn't - my only concern would be whether `apierrors.IsNotFound(err)` will recognize the wrapped error that gets tossed (via `errors.WithStack(err)`).